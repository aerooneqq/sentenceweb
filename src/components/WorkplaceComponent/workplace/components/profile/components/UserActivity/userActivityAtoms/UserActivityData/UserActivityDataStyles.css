.userActivityData { 
    width: 100%;
    height: 100%;

    background: transparent;
}

.userActivityDataHeaderCont { 
    width: 100%;
    height: auto;

    box-sizing: border-box;

    padding-left: 5%;
    padding-right: 5%;

    display: flex;
    flex-direction: row;

    justify-content: center;
    align-items: center;
}

.userActivityDataHeaderText { 
    font-size: 14px;
    color: black;
}

.userActivityDataFill { 
    flex-grow: 1;
}

.closeUserActivityIcon { 
    width: 20px;
    height: 20px;

    margin-top: 5px;
}