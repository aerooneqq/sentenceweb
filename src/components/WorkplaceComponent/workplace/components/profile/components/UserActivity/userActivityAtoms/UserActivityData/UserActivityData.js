import React, {Component} from "react"

//Styles
import "./UserActivityDataStyles.css"

//Icons
import closeIconStatic from "./img/close_user_activity_data_icon.png";
import closeIconActive from "./img/close_user_activity_data_icon_active.png"

export default class UserActivityData extends Component  { 

    constructor(props) { 
        super(props);

        this.state = { 
            isCloseIconActive: false
        };

        this.onCloseIconMouseEnter = this.onCloseIconMouseEnter.bind(this);
        this.onCloseIconMouseLeave = this.onCloseIconMouseLeave.bind(this);
    }

    onCloseIconMouseEnter() { 
        this.setState({ 
            isCloseIconActive: true
        });
    }

    onCloseIconMouseLeave() { 
        this.setState({ 
            isCloseIconActive: false
        });
    }
    
    render() { 
        return ( 
            <div className = "userActivityData">
                <div className = "userActivityDataHeaderCont">
                    <div className = "userActivityDataHeaderText">
                        Activity data
                    </div>
                    <div className = "userActivityDataFill" />
                    <div className = "userActivityDataHeaderIcon">
                        <img src = {this.state.isCloseIconActive === true ? closeIconActive : closeIconStatic } 
                             alt = "Close"
                             className = "closeUserActivityIcon"
                             onMouseEnter = {this.onCloseIconMouseEnter}
                             onMouseLeave = {this.onCloseIconMouseLeave}
                             onClick = {this.props.closeActivityData} />
                    </div>
                </div>
            </div>
        )
    }
}