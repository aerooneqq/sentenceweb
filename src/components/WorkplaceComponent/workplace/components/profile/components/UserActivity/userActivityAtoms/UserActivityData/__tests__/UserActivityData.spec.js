import React from "react"; 
import {shallow} from "enzyme"; 
import {shallowToJson} from "enzyme-to-json";
import UserActivityData from "../UserActivityData";

const closeActivitySpy = jest.fn(() => {});
const props = { 
    activityData: {
        activities: []
    },
    closeActivityData: closeActivitySpy
}

describe("UserActivityData", () => { 
    it("should render correctly", () => { 
        expect(shallowToJson(shallow(<UserActivityData {...props}/>))).toMatchSnapshot();
    });

    it("should have proper intial state", () => { 
        const userActivityData = shallow(<UserActivityData {...props}/>);

        const properState = { 
            isCloseIconActive: false
        }

        for (let key in properState) { 
            expect(properState[key]).toBe(userActivityData.state(key));
        }
    });

    it("should change isCloseIconActive to true after mouse enter is called", () => { 
        const userActivityData = shallow(<UserActivityData {...props} />);
        userActivityData.find("img").simulate("mouseEnter");

        expect(userActivityData.state("isCloseIconActive")).toBe(true);
    });

    it("should change isCloseIconActive to false whren the mouse leave event is called", () => { 
        const userActivityData = shallow(<UserActivityData {...props}/>);
        userActivityData.find("img").simulate("mouseLeave");

        expect(userActivityData.state("isCloseIconActive")).toBe(false);
    });

    it("should call closeActivityData when img is clicked", () => { 
        const userActivityData = shallow(<UserActivityData {...props}/>);
        userActivityData.find("img").simulate("click");

        expect(closeActivitySpy).toHaveBeenCalledTimes(1);
    })
})