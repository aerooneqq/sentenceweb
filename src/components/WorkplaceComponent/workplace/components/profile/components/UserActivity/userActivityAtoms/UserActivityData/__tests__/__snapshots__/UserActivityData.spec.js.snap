// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserActivityData should render correctly 1`] = `
<div
  className="userActivityData"
>
  <div
    className="userActivityDataHeaderCont"
  >
    <div
      className="userActivityDataHeaderText"
    >
      Activity data
    </div>
    <div
      className="userActivityDataFill"
    />
    <div
      className="userActivityDataHeaderIcon"
    >
      <img
        alt="Close"
        className="closeUserActivityIcon"
        onClick={[MockFunction]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        src="close_user_activity_data_icon.png"
      />
    </div>
  </div>
  <div
    className="singleUserActivitiesCont"
  />
</div>
`;
